object-store-client-common/src/main/scala/uk/gov/hmrc/objectstore/client/model/http/ObjectStoreWrite.scala

```diff
@@ -16,15 +16,22 @@
 
 package uk.gov.hmrc.objectstore.client.model.http
 
-import scala.language.higherKinds
+import uk.gov.hmrc.objectstore.client.model.Monad
 
-trait ObjectStoreWrite[BODY, REQ] {
-  def write(body: BODY): REQ
-}
+case class Payload[CONTENT](length: Long, md5Hash: String, content: CONTENT)
```

Comment:
This has been moved to common but isn't a part of the HttpClient trait (used from play subproject). I wonder if this and HttpBody are actually the same thing? (I think we convert from a Payload to a HttpBody just to write it).

The thought behind length being optional in HTTP body is that Play works it out for all non-chuncked requests. MD5 was optional as it wasn't implemented yet (it probably shouldn't be optional on HTTPBody)